src/transformers/models/prophetnet/modeling_prophetnet.py

```diff
@@ -1397,7 +1397,7 @@ def forward(
         # prepare encoder attention mask
         if encoder_attention_mask is not None:
             extended_encoder_attention_mask = (
-                1.0 - encoder_attention_mask[:, None, :].repeat(self.config.num_attention_heads, 1, 1)
+                1.0 - encoder_attention_mask[:, None, :].repeat(self.config.num_decoder_attention_heads, 1, 1)
```

Comment:
The encoder_attention_mask will be added to the cross-attention weights which also rely on `num_decoder_attention_heads`